manual/core/statements/prepared/README.md

```diff
@@ -55,35 +55,69 @@ client                            driver                Cassandra
+
+If you have a unique query that is executed only once, a [simple statement](../simple/) will be more
+efficient. But note that this should be pretty rare: most client applications typically repeat the
+same queries over and over, and a parameterized version can be extracted and prepared.  
+
+### Preparing
+
+`Session.prepare()` accepts either a plain query string, or a `SimpleStatement` object. If you use a
+`SimpleStatement`, its execution parameters will propagate to bound statements:
+
+```java
+SimpleStatement simpleStatement =
+    SimpleStatement.builder("SELECT * FROM product WHERE sku = ?")
+        .withConsistencyLevel(DefaultConsistencyLevel.QUORUM)
+        .build();
+PreparedStatement preparedStatement = session.prepare(simpleStatement);
+BoundStatement boundStatement = preparedStatement.bind();
+assert boundStatement.getConsistencyLevel() == DefaultConsistencyLevel.QUORUM;
+``` 
+
+For more details, including the complete list of attributes that are copied, refer to
+[API docs][Session.prepare].
+
 The driver caches prepared statements: if you call `prepare()` multiple times with the same query
-string (or a `SimpleStatement` with the same execution characteristics), you will get the same
-`PreparedStatement` instance. We still recommend keeping a reference to it (for example by caching
-it as a field in a DAO); if that's not possible (e.g. if query strings are generated dynamically),
-it's OK to call `prepare()` every time: there will just be a small performance overhead to check the
-internal cache. Note that caching is based on:
+string (or a `SimpleStatement` with the same execution parameters), you will get the same
+`PreparedStatement` instance:
+ 
+```java
+PreparedStatement ps1 = session.prepare("SELECT * FROM product WHERE sku = ?");
+// The second call hits the cache, nothing is sent to the server:
+PreparedStatement ps2 = session.prepare("SELECT * FROM product WHERE sku = ?");
+assert ps1 == ps2;
+``` 
+ 
+We still recommend avoiding repeated calls to `prepare()`; if that's not possible (e.g. if query
+strings are generated dynamically), there will just be a small performance overhead to check the
+cache on every call.
+
+Note that caching is based on:
 
 * the query string exactly as you provided it: the driver does not perform any kind of trimming or
   sanitizing.
 * all other execution parameters: for example, preparing two statements with identical query strings
-  but different consistency levels will yield distinct prepared statements.
+  but different consistency levels will yield two distinct prepared statements (that each produce
+  bound statements with their respective consistency level).
 
 The size of the cache is exposed as a session-level [metric](../../metrics/)
 `cql-prepared-cache-size`. The cache uses [weak values]([guava eviction]) eviction, so this
 represents the number of `PreparedStatement` instances that your application has created, and is
 still holding a reference to.
```

Comment:
Maybe we can add info that by default the size of the cache in unbounded?

Reply:
The thing is, if we mention that, it plants the idea that the cache could be bounded. And I'm not sure how that would work:

Let's say the cache has a capacity of 10 with an LRU eviction strategy. You prepare statement number 11, which means another statement gets evicted, let's stay number 1. But all that means is that the next `session.prepare` call for statement 1's query won't hit the cache. The application might still hold a reference to `preparedStatement1`, and executing it might still work if it's not been evicted server-side. The only thing we're achieving is that if the query is prepared again, it will hit the server (unnecessarily) and the driver will build a duplicate `PreparedStatement` instance. So we might be wasting heap space instead of saving it.

Unbounded with weak values works well because it adapts to what the application is preparing and holding onto. Of course if you have a rogue application that prepares in a loop and keeps a reference to every statement, it won't work well, but neither will a bounded cache from the explanations above.